### PR TITLE
Manage user profile

### PR DESCRIPTION
**Pull request by ioannanedelcu**
Add authentication feature
Store user's info and display them on profile page

### CODE REVIEW

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -16,13 +16,33 @@
 window.onscroll = function () {
   stickyNavbar();
 };
-
 function stickyNavbar() {
   var navbar = document.getElementById("navbar");
+  var initialNavBarOffset = 308;
 
   if (window.pageYOffset >= 308) {
```
Comment (nd0761) [inline src/main/webapp/script.js]: I think you meant not 308 but initialNavBarOffset?

Comment (nd0761) [reply src/main/webapp/script.js]: my bad, didn't notice that it was deleted.

File: src/main/java/com/google/sps/data/User.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+public final class User {
+
+  private final String fname;
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/data/User.java:19]: Please use more descriptive variable named (eg firstName) throughout.

File: src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// ** Returns the login status of a user */
+@WebServlet("/login")
+public class LogInServlet extends HttpServlet {
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/servlets/LogInServlet.java:28]: Casing is inconsistent for login. Please use `Login` or `LogIn` consistently.

File: src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  static class LoginStatus {
+    boolean loggedIn;
+    String actionURL;
+
+    LoginStatus(boolean loggedIn, String actionURL) {
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/servlets/LogInServlet.java:33]: `actionURL` is not consistent with the java naming conventions. Should be `actionUrl`. See https://google.github.io/styleguide/javaguide.html#s5.3-camel-case

Comment (ioannanedelcu) [reply src/main/java/com/google/sps/servlets/LogInServlet.java:33]: Done

File: src/main/java/com/google/sps/servlets/LogInServlet.java (hunk 3 above)
Comment (ioannanedelcu) [reply src/main/java/com/google/sps/servlets/LogInServlet.java:28]: Done

File: src/main/java/com/google/sps/data/User.java (hunk 2 above)
Comment (ioannanedelcu) [reply src/main/java/com/google/sps/data/User.java:19]: Done

Comment (conradgrobler) [reply src/main/java/com/google/sps/data/User.java:19]: It seems that there is a missing commit where these changes have been done that has not yet been pushed to the server.

Comment (ioannanedelcu) [reply src/main/java/com/google/sps/data/User.java:19]: Yes, I just have a few a problems with my cloud project right now. I hope I'll fix them soon and then add the commit.

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 5]
```diff
@@ -43,8 +43,8 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
     // Get the input from the form.
-    String fname = getParameter(request, "fname", "Not set");
-    String lname = getParameter(request, "lname", "Not set");
+    String fname = getParameter(request, "first -name", "Not set");
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:46]: The space in the parameter name will probably cause an issue.

Comment (conradgrobler) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:46]: Also please change these local variable names to be more descriptive.

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+@WebServlet("/user-info")
+@SuppressWarnings("serial")
+public class ProfileInfoServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    // Check if user is logged in.
+    if (!userService.isUserLoggedIn()) {
+      System.out.println("ERROR:You are not logged in!");
```
Comment (ipetr0v) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:42]: Who will read this message?
It it somehow outputed to the client?
If so, how will he see it? Clients don't look into JS console

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    // Get the input from the form.
+    String firstName = getParameter(request, "first-name", "Not set");
+    String lastName = getParameter(request, "last-name", "Not set");
+    String nickname = getParameter(request, "nickname", "Anonym");
+    boolean notifications;
+    if ((getParameter(request, "radio", "mute")) == "mute") {
+      notifications = false;
+    } else {
+      notifications = true;
+    }
```
Comment (ipetr0v) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:55]: Is it possible to do the following:
```java
boolean notifications = getParameter(request, "radio", "mute") != "mute"
```

**Pull request merged**